10.1/s2i/bin/assemble

```diff
@@ -275,6 +275,22 @@ if [ -f "$LOCAL_SOURCE_DIR/pom.xml" ]; then
   if [ -n "$MAVEN_ARGS_APPEND" ]; then
     export MAVEN_ARGS="$MAVEN_ARGS $MAVEN_ARGS_APPEND"
   fi
+
+  CONTAINER_HEAP_PERCENT=${CONTAINER_HEAP_PERCENT:-0.50}
+  CONTAINER_MEMORY_IN_BYTES=`cat /sys/fs/cgroup/memory/memory.limit_in_bytes`
+  CONTAINER_MEMORY_IN_MB=$((${CONTAINER_MEMORY_IN_BYTES}/1024**2))
+  CONTAINER_HEAP_MAX=$(echo "${CONTAINER_MEMORY_IN_MB} ${CONTAINER_HEAP_PERCENT}" | awk '{ printf "%d", $1 * $2 }')
+  JAVA_HEAP_MAX=-Xmx${CONTAINER_HEAP_MAX}
```

Comment:
What was the thought process around not gating this by `DEFAULT_MEMORY_CEILING=$((2**40-1))` and `if [ "${CONTAINER_MEMORY_IN_BYTES}" -lt "${DEFAULT_MEMORY_CEILING}" ]; then
`